src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -96,6 +97,34 @@ public function getDefaultData()
                 'wholesale_price' => 0,
                 'unit_price' => 0,
             ],
+            'shipping' => [
+                'width' => 0,
+                'height' => 0,
+                'depth' => 0,
+                'weight' => 0,
+            ],
+        ];
+    }
+
+    /**
+     * @param ProductForEditing $productForEditing
+     *
+     * @return array<string, mixed>
+     */
+    private function extractShippingData(ProductForEditing $productForEditing): array
```

Comment:
I like the function to extract the different parts, I think we should do it for `basic` and `price` as well
Do you think you can change it in this PR? or it is worth doing in another (very small) PR?